logReader/src/main/java/org/corfudb/logReader/logReader.java

```diff
@@ -217,8 +217,11 @@ public final void printLogEntry(final LogEntry entry, final boolean showBinary)
             case TRIMMED:
                 System.out.println("DataType: TRIMMED");
                 break;
+            case CHECKPOINT:
+                System.out.println("DataType: CHECKPOINT");
+                break;
             default:
-                System.out.println("UNKNOWN DataType");
+                System.out.printf("UNKNOWN DataType %s\n", dt.toString());
```

Comment:
remove toString() - it is not necessary, and if the value is null, your code will throw NPE.